vm/src/stdlib/thread.rs

```diff
@@ -137,6 +138,26 @@ pub(crate) mod _thread {
             Ok(())
         }
 
+        #[pymethod]
+        fn _at_fork_reinit(&self, _vm: &VirtualMachine) -> PyResult<()> {
+            if self.mu.is_locked() {
+                unsafe {
+                    self.mu.unlock();
+                };
+            }
+            // Casting to AtomicCell is as unsafe as CPython code.
+            // Using AtomicCell will prevent compiler optimizer move it to somewhere later unsafe place.
+            // It will be not under the cell anymore after init call.
+
+            let new_mut = RawMutex::INIT;
+            unsafe {
+                let old_mutex: AtomicCell<&RawMutex> = AtomicCell::new(&self.mu);
+                old_mutex.swap(&new_mut);
```

Comment:
Isn't this swap only the reference which `old_mutex` holds?

Is this working?
```suggestion
                let old_mutex: &AtomicCell<RawMutex> = std::mem::transmute(&self.mu);
                old_mutex.swap(new_mut);
```

Reply:
This complies without any issue so looks like its working.
Just to reconfirm, 
i.e. if mu is mutex and mu_ptr is reference to mu
then in original code:
old_mu_ptr = copy(mu_ptr)
old_mu_ptr = new_mut

after above changes:
old_mu_ptr = mu_ptr
old_mu_ptr = new_mut

Am I getting it right?

Reply:
`AtomicCell::swap` replaces `T`. `T` is `&RawMutex` here. `self.mu` is `RawMutex`. It can't be changed.

How do you know this is working?

Reply:
> How do you know this is working?

I tried something like this
```python
import os, time
import threading
from threading import Lock

l = Lock()
l.acquire()

def fork_in_thread():
    print("In fork in thread")

    pid = os.fork()
    if pid:
        # parent
        time.sleep(5)
        os._exit(0)

    # child process
    l = Lock()
    print("Child acquiring lock") 
    l.acquire() # lock from main thread shouldn't block this
    print(f'Child: lock acquired: {l.locked()}') 


t= threading.Thread(target=fork_in_thread, args=() )
t.run()
# block the main thread: fork_in_thread() exit the process
time.sleep(10)
```
This is based on https://bugs.python.org/issue6721
This gave results in cpython and current branch of rustpython
```
In fork in thread
Child acquiring lock
Child: lock acquired: True
```

For main branch of rustpython, it gave this:
```
In fork in thread
Exception ignored in: <function _after_fork at 0x5630acf32170>
  File "/home/ankit/tt.py", line 25, in <module>
    t.run()
  File "/home/ankit/RustPython/pylib/Lib/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ankit/tt.py", line 11, in fork_in_thread
    pid = os.fork()
AttributeError: 'lock' object has no attribute '_at_fork_reinit'
Child acquiring lock
Child: lock acquired: True

```
It actually work fine in main branch sans missing function definition `_at_fork_reinit` which doesn't adds anything.